enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,398 @@
+  - “@stbenjam”
+  - “@dustymabe”
+  - “@miabbott”
+  - “@dougbtv”
+  - “@eparis”
+  - “@markmc”
+  - “@jwforres”
+creation-date: 2020-06-30
+last-updated: 2020-06-30
+status: implemented
+see-also:
+  - "/enhancements/rhcos/static-networking-enhancements.md"
+  - "/enhancements/network/20190903-SRIOV-GA.md"
+  - "/enhancements/network/host-level-openvswitch.md"
+  - "/enhancements/baremetal/baremetal-provisioning-config.md"
+  - "/enhancements/baremetal/baremetal-provisioning-optional.md"
+---
+
+# Host Network Configuration
+
+## Summary
+
+This document serves as a high level overview of host network configuration for
+OpenShift 4.x clusters.  Host network configuration spans multiple features in
+multiple components.  This enhancement aims to describe use cases and point to
+various references for current or future solutions that help satisfy those use
+cases.
+
+## Motivation
+
+Configuring host network devices on an OpenShift cluster looks a bit different
+than a standalone Linux host.  This document aims to be a starting reference
+for learning more about how this works right now and where to look to learn
+more.
+
+### Goals
+
+* Document the current configuration interfaces available for host networking
+* Provide references to other proposals
+
+### Non-Goals
+
+* Proposing new enhancements.  Instead, those should be covered in other
+  enhancements that this one may reference.
+
+
+## Use Cases
+
+The purpose of this section is not to enumerate every possible use case, but it
+should cover enough use cases to demonstrate the range and flexibility of host
+network configuration needed in OpenShift.
+
+This section could use several more additions - contributions welcome!
+
+### Use Case #1 - Bare Metal, Primary Network VLAN
+
+* Bare metal servers with a single network interface
+* We desire PXE booting these servers on the untagged network for provisioning
+* We must configure a VLAN on this interface to access the primary external
+  network for the cluster
+* Access to this network must be configured very early, before ignition will be
+  able to reach the machine config server
+
+### Use Case #2 - Extra VLAN, Post Install Configuration
+
+* After cluster installation, configure another VLAN on a network interface
+  that specific workloads will need access to via multus
+
+### Use Case #3 - Static Network Configuration Only
+
+* Must be able to override default RHCOS behavior of requiring DHCP to be
+  present on at least one interface
+* Must be able to configure static network configuration for every interface
+  used by the cluster
+
+
+## Existing Configuration Interfaces
+
+### Summary
+
+First, here’s a summary of how an administrator may configure host networking
+in a cluster.  More details about each interface are in the following sections.
+
+#### First Boot
+
+When an RHCOS host first boots, the initramfs environment must be able to reach
+the Machine Config Server for Ignition to succeed.  By default, dracut will
```

Comment:
Maybe it doesn't belong in this document, but where does the MCS come from? Also, once networking is available, how does the host find the MCS? Did we have to tell it manually at boot time or is it indicated in the DHCP response or what?

Reply:
I added the best reference i could find that describes what the MachineConfigServer is.  I also added a note here that clarifies that the MCS location is via URL.

Reply:
Sorry, not what I meant by "where does it come from". I meant, where is the MCS running before they install the first node?

Reply:
Ah ok, I'll add a note that clarifies that the MCS starts on the boostrap host while the first nodes are being installed